### PR TITLE
[#192] Add support for read/write sequence/json/avro/zip/gzip file formats from HDFS

### LINKED ISSUE #192: [Enhancement] Add support for read/write sequence/json/avro/zip/gzip file formats from HDFS
**What feature are your requesting?**
Add support for read/write sequence/json/avro/zip/gzip file formats from HDFS.

**What benefits do you foresee from the feature you are requesting?**
Adding more usability of HDFS API.

**Potential solution/ideas?**
Add the respective logic based on gimel.hdfs.data.format property.

**Additional context**

### PR DESCRIPTION
**Pull request by anishanainani**
Make sure you have checked all steps below.

### GitHub Issue
Fixes #192 

### Checklist:

- [x] This pull request updates the documentation
- [x] This pull request changes library dependencies
- [x] Title of the PR is of format (example) : [#25][Github] Add Pull Request Template

### What is the purpose of this pull request?
- Refactor HDFS API
- Add additional file formats support to HDFS API

### How was this change validated?
Spark shell

Examples:
```
val options = Map("rest.service.method" -> "https",
"rest.service.host" -> "udc-service-host",
"rest.service.port" -> "443",
"gimel.hdfs.data.format" -> "csv",
"gimel.hdfs.nn" -> "hdfs://cluster-hdfs.example.com:8020/",
"gimel.hdfs.data.location" -> "hdfs://cluster/user/hadoopuser/seq_test_csv")

val df = dataset.read("Hdfs.Cluster.default.cluster_user_hadoopuser_seq_test_csv", options)
```

### Commit Guidelines
- [x] My commits all reference GH issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala [hunk 1]
```diff
@@ -19,97 +19,94 @@
...
   /** Read Implementation for HDFS DataSet
     *
-    * @param dataset Name of the PCatalog Data Set
+    * @param dataset Name of the UDC Data Set
     * @param dataSetProps
     *                props is the way to set various additional parameters for read and write operations in DataSet class
     *                Example Usecase : to get 10 factor parallelism (specifically)
     *                val props = Map("coalesceFactor" -> 10)
-    *                val data = Dataset(sc).read("flights", props)
+    *                val data = Dataset(sc).read("wuser", props)
```
Comment (Dee-Pac) [inline gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala]: @anishanainani  - Could you pls remove `wuser` and make it generic?

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala [hunk 2]
```diff
@@ -150,7 +146,7 @@ class DataSet(sparkSession: SparkSession) extends GimelDataSet(sparkSession: Spa
     *                Dataset(sc).write(clientDataFrame, props)
     * @return RDD[T]
     */
-  def write[T: TypeTag](dataset: String, rdd: RDD[T], dataSetProps: Map[String, Any]): RDD[T] = {
+  def write[T: universe.TypeTag](dataset: String, rdd: RDD[T], dataSetProps: Map[String, Any]): RDD[T] = {
```
Comment (Dee-Pac) [inline gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala]: NIT: do we need to quality `universe.TypeTag` ? Could you please elaborate what issue you faced requiring you to qualify `universe.`?

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/conf/HdfsClientConfiguration.scala [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/*
```
Comment (Dee-Pac) [inline gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/conf/HdfsClientConfiguration.scala]: Should this file go back under @bthilakraj 's commit?

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala [hunk 4]
```diff
@@ -160,60 +156,14 @@ class DataSet(sparkSession: SparkSession) extends GimelDataSet(sparkSession: Spa
     }
   }
 
-  /**
-    * Function reads  cross cluster data and return as a dataframe
-    * @param pathToRead             Path of HDFS Data
-    * @param crossClusterDataFormat Cross Cluster Format
-    * @return DataFrame
-    */
-  def readAcrossCluster(pathToRead: String, crossClusterDataFormat: String, inferSchema: String, header: String): DataFrame = {
```
Comment (Dee-Pac) [inline gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala]: It seems we are removing `crosscluster` functionality. Is it being moved elsewhere or is it completely removed? I've not gotten to the rest of the files as I'm writing this comment.

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala (hunk 2 above)
Comment (anishanainani) [reply gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala]: @Dee-Pac It was part of the sync. We can remove it.

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/conf/HdfsClientConfiguration.scala (hunk 3 above)
Comment (anishanainani) [reply gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/conf/HdfsClientConfiguration.scala]: @Dee-Pac This was added by abhishek. He separated out the configs in HDFS.

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala (hunk 4 above)
Comment (anishanainani) [reply gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala]: As part of the HDFS refactor, @AbhishekChinni figured out that the logic for read with different formats is same with/without cross cluster. Its just hdfs namenode url which matters of which cluster to connect to. Thats why he added a single method readPath which resolves the format and runs the required logic.

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala (hunk 1 above)
Comment (anishanainani) [reply gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala]: Done.

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala (hunk 4 above)
Comment (Dee-Pac) [reply gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/DataSet.scala]: Thanks for the clarification.

File: gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/conf/HdfsClientConfiguration.scala (hunk 3 above)
Comment (Dee-Pac) [reply gimel-dataapi/gimel-connectors/gimel-hive-1.2/src/main/scala/com/paypal/gimel/hdfs/conf/HdfsClientConfiguration.scala]: 👍

**Pull request merged**